ecosystem/purescript-0.14-migration-guide.md

```diff
@@ -0,0 +1,622 @@
+
+The `Semigroup` instance for `Map` will be restored in PureScript 0.15, but will be unbiased rather than left-biased. The instance is removed in PureScript 0.14 to avoid silently changing the behavior of existing code. Since the `Monoid` class requires `Semigroup`, the `Monoid` instance for `Map` has also been removed.
+
+To mitigate breaking changes, a new `SemigroupMap` newtype has been added to the `Data.Map` module which supplies an unbiased `Semigroup` instance.
+
+**To fix:**
+
+If you previously relied on the `Monoid` instance for `Data.Map` (ie. you used `mempty`), then you may be able to simply update your code to use `Map.empty` directly instead:
+
+```diff
+  import Data.Map as Map
+
+  m :: Map Int String
+- m = mempty
++ m = Map.empty
+```
+
+If this is not possible and you previously relied on the `Semigroup` or `Monoid` instances for `Data.Map`, then you can use the `SemigroupMap` newtype to continue using those instances. Note that the behavior of the instances has changed to be unbiased rather than left-biased. You can use the `First` and `Last` newtypes to produce left-biased or right-biased behavior, as demonstrated below.
+
+```purs
+import Data.Map (SemigroupMap, singleton)
+import Data.Semigroup.First (First(..))
+import Data.Semigroup.Last (Last(..))
+
+-- Pseudocode demonstrating evaluation
+let
+  s :: forall key value. key -> value -> SemigroupMap key value
+  s k v = SemigroupMap (Data.Map.singleton k v)
+
+(s 1     "foo") <> (s 1     "bar") == (s 1  "foobar")
+(s 1 (First 1)) <> (s 1 (First 2)) == (s 1 (First 1))
+(s 1  (Last 1)) <> (s 1  (Last 2)) == (s 1  (Last 2))
+```
+
+#### The `Semigroup` instance for foreign objects has changed.
+
+Implemented in [purescript-foreign-object#23](https://github.com/purescript/purescript-foreign-object/pull/23) and [purescript-foreign-object#19](https://github.com/purescript/purescript-foreign-object/pull/19).
+
+The `Semigroup` instance for `Foreign.Object.Object` previously appended values from the right to values from the left, which is in reverse order from how this instance should have been implemented. To demonstrate:
+
+```purs
+-- these were previously equivalent
+singleton k a <> singleton k b == singleton k (b <> a)
+
+-- now these are equivalent, as expected
+singleton k a <> singleton k b == singleton k (a <> b)
+```
+
+Code using these instances will now append in the correct order.
+
+**To fix:**
+
+Most users should not change their code and will now be using the correct instance. However, if you were intentionally relying on the behavior of this instance then you can recover it by writing a newtype around `Object` and implementing this newtype instance as desired. For example:
+
+```purs
+newtype Reversed v = Reversed (Object v)
```

Comment:
I'd suggest using `Object (Dual a)` since we already have a `Dual` newtype in prelude which does what we want here: https://pursuit.purescript.org/packages/purescript-prelude/4.1.1/docs/Data.Monoid.Dual#t:Dual

Reply:
I was wondering whether we already had a newtype for this exact purpose.